doc/support.rst

```diff
@@ -2,96 +2,54 @@
-  where updates to the main repository and test failures get notified.
-
-
-.. _user_questions:
-
-User questions
+User Questions
 ==============
 
-- Some scikit-learn developers support users on StackOverflow using
-  the `[scikit-learn] <https://stackoverflow.com/questions/tagged/scikit-learn>`_
-  tag.
-
-- For general theoretical or methodological Machine Learning questions
-  `stack exchange <https://stats.stackexchange.com/>`_ is probably a more
-  suitable venue.
-
-In both cases please use a descriptive question in the title field (e.g.
-no "Please help with scikit-learn!" as this is not a question) and put
-details on what you tried to achieve, what were the expected results and
-what you observed instead in the details field.
-
-Code and data snippets are welcome. Minimalistic (up to ~20 lines long)
-reproduction script very helpful.
+Engage with the scikit-learn community and seek answers to your questions:
 
-Please describe the nature of your data and how you preprocessed it:
-what is the number of samples, what is the number and type of features
-(i.d. categorical or numerical) and for supervised learning tasks,
-what target are your trying to predict: binary, multiclass (1 out of
-``n_classes``) or multilabel (``k`` out of ``n_classes``) classification
-or continuous variable regression.
+- **StackOverflow**: Ask questions tagged with `[scikit-learn] <https://stackoverflow.com/questions/tagged/scikit-learn>`_.
+- **General Machine Learning Queries**: For broader machine learning discussions, visit `Stack Exchange <https://stats.stackexchange.com/>`.
 
-User questions should **not be asked on the bug tracker**, as it crowds
-the list of issues and makes the development of the project harder.
+When posting questions:
+- Use descriptive titles.
+- Provide detailed context, expected results, and actual observations.
+- Include code and data snippets (preferably minimalistic scripts, up to ~20 lines).
+- Describe your data and preprocessing steps, including sample size, feature types (categorical or numerical), and the target for supervised learning tasks (classification type or regression).
 
-.. _bug_tracker:
+**Note**: Avoid asking user questions on the bug tracker to keep the focus on development.
 
-Bug tracker
+Bug Tracker
 ===========
 
-If you think you've encountered a bug, please report it to the issue tracker:
+Encountered a bug? Report it on our issue tracker:
 
-https://github.com/scikit-learn/scikit-learn/issues
+`Scikit-learn Issues <https://github.com/scikit-learn/scikit-learn/issues>`
 
-Don't forget to include:
+Include in your report:
+- Steps or scripts to reproduce the bug.
+- Expected and observed outcomes.
+- Python or gdb tracebacks, if applicable.
 
-- steps (or better script) to reproduce,
+Enhance your report with a link to a `Gist <https://gist.github.com>`_ containing a standalone script and a sample of your dataset (use `numpy.savetxt` for exporting CSV files).
```

Comment:
Instead of gist, we now encourage [crafting a minimal reproducer for scikit-learn](https://scikit-learn.org/stable/developers/minimal_reproducer.html). I think it's worth linking it here.

Remember you can use the sphinx [cross-referencing syntax](https://www.sphinx-doc.org/en/master/usage/referencing.html).